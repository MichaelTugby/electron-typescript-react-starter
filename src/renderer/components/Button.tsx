/** @jsx jsx */
import { jsx, css } from "@emotion/react";
import PropTypes from "prop-types";

import type { FunctionComponent } from "react";

interface ButtonTypes {
  children: string;
}
const Button: FunctionComponent<ButtonTypes> = ({ children }) => (
  <button
    css={css`
      color: blue;
      &:hover {
        color: red;
      }
    `}
  >
    {children}
  </button>
);
Button.propTypes = {
  children: PropTypes.string.isRequired,
};

export default Button;
