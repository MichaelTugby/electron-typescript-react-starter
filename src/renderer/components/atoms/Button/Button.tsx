/** @jsx jsx */
import { css, jsx, useTheme } from "@emotion/react";
import type { ButtonHTMLAttributes, FunctionComponent } from "react";

interface IButtonTypes extends ButtonHTMLAttributes<HTMLButtonElement> {
  primary?: boolean;
}
const Button: FunctionComponent<IButtonTypes> = ({
  children,
  primary = false,
  ...props
}) => {
  const theme = useTheme();
  return (
    <button
      className="button"
      css={css`
        background-color: ${primary ? theme.colors.primary : "black"};
        padding: 5px 10px;
        border: 1px solid black;
        position: relative;
        cursor: pointer;
        &:after {
          content: "";
          position: absolute;
          height: 100%;
          width: 100%;
          background-color: white;
          left: 0;
          top: 0;
          opacity: 0;
        }
        &:hover:after {
          opacity: 0.3;
        }
        &:focus {
          outline: 0;
        }
      `}
      {...props}
    >
      {children}
    </button>
  );
};

export default Button;
