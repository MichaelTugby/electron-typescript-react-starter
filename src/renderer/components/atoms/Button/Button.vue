<template>
  <button class="button" :class="{ 'button--primary': primary }">
    <slot />
  </button>
</template>

<script lang="ts">
import { defineComponent } from "vue";
export default defineComponent({
  props: {
    primary: Boolean,
  },
});
</script>

<style lang="scss" scoped>
.button {
  &--primary {
    background-color: var(--app-primary);
  }
  padding: 5px 10px;
  border: 1px solid black;
  position: relative;
  cursor: pointer;
  &:after {
    content: "";
    position: absolute;
    height: 100%;
    width: 100%;
    background-color: white;
    left: 0;
    top: 0;
    opacity: 0;
  }
  &:hover:after,
  &:focus:after {
    opacity: 0.3;
    outline: 0;
  }
}
</style>
