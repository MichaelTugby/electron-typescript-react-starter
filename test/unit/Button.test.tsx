import React from "react";
import { cleanup, render } from "@testing-library/react";
import Button from "~/components/Button";

describe("Button Component", () => {
  afterEach(cleanup);

  it("snapshot has not changed", () => {
    const { container } = render(<Button>Test</Button>);
    expect(container).toMatchSnapshot();
  });
});
