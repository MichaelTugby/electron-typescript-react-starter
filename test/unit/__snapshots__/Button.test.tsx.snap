// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button Component snapshot has not changed 1`] = `
<div>
  <button
    class="css-24dw16-Button"
  >
    Test
  </button>
</div>
`;
